src/CodegenFunctionish.hack

```diff
@@ -53,6 +54,25 @@ abstract class CodegenFunctionish implements ICodeBuilderRenderer {
     return $this;
   }
 
+  public function setContexts(
+    ?string $contexts,
+  ): this {
+    if($contexts !== null) {
+      return $this->setContextsf('%s', $contexts);
+    } else {
+      $this->contexts = null;
+      return $this;
+    }
+  }
+
+  public function setContextsf(
+    Str\SprintfFormatString $contexts,
+    mixed ...$args
+  ): this {
+    $this->contexts = \vsprintf($contexts, $args);
+    return $this;
+  }
```

Comment:
I don't think something like this should be needed, especially if it takes a keyset<string> - if there's a need for this, could you give an example?